.github/workflows/checks.yml

```diff
@@ -0,0 +1,96 @@
+name: Basic checks
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  # Only run once for latest commit per ref and cancel other (previous) runs.
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-code-style:
+    name: Check Code Style
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      - name: Set up JDK 11
+        uses: olafurpg/setup-scala@v10
```

Comment:
This action deprecated de-facto (see https://github.com/olafurpg/setup-scala/issues/49)
In Play we use a [Coursier/Setup Action](https://github.com/coursier/setup-action) (see https://github.com/playframework/.github/pull/2/files as example) 😉

Reply:
Ok, I will adapt. Thanks for the heads-up